src/network/ip_address.cpp

```diff
@@ -40,6 +40,10 @@ std::array<uint8_t, 4> parse(const std::string& ip)
     int b = -1;
     int c = -1;
     int d = -1;
+
+    if (ip.empty())
+        return {{static_cast<uint8_t>(a), static_cast<uint8_t>(b), static_cast<uint8_t>(c), static_cast<uint8_t>(d)}};
+
```

Comment:
I don't see why we need to special case this here.

The caller should instantiate an IPAddress with 0xFFFFFFFF instead, otherwise it should throw an exception.